### PR TITLE
Fix #3474: Merge faq single activity into single layout file.

### LINKED ISSUE #3474: Convert faq_single_activity into single layout file
Currently there are 3 versions of `faq_single_activity .xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in the `dimens.xml` file and use it accordingly.

Note: In PR, make sure you add before and after screenshots of `mobile-portrait`, `mobile-landscape`, `tablet-portrait`, and `tablet-landscape` for comparison and make sure that there is no difference between before and after UI.

### PR DESCRIPTION
**Pull request by rishidyno**
## Explanation

Fix #3474

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rishidyno** [comment]
|hd|dds|
|-------|-------|
| ddc| sss|
|ds|fae|

**Comment by rishidyno** [comment]
| OLD Screenshots         | NEW Screenshots |
| -----------       | -----------      |
| Screenshot_20210727_025302 | Screenshot_20210727_024438 |
| Screenshot_20210727_025318 | Screenshot_20210727_024330 |
| Screenshot_20210727_025340 | Screenshot_20210727_024527 | 
|Screenshot_20210727_025333 | Screenshot_20210804_130410 |

**Comment by rishidyno** [comment]
@rt4914 I have added the required screenshots and all tests have passed. Please, review my PR

File: app/src/main/res/values-sw600dp-land/booleans.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+  <bool name="faq_single_activity_scroll_fill_view_port">true</bool>
+</resources>
```
Comment (prayutsu) [inline app/src/main/res/values-sw600dp-land/booleans.xml]: Add EOF

**Review by prayutsu — COMMENTED** [review]
Generally LGTM, just have one comment.

File: app/src/main/res/values-sw600dp-land/booleans.xml (hunk 1 above)
Comment (rishidyno) [reply app/src/main/res/values-sw600dp-land/booleans.xml]: @prayutsu I have added EOF please review it.

**Comment by rishidyno** [comment]
@rt4914 @anandwana001 Please review my PR.

**Comment by prayutsu** [comment]
@rishidyno PR mostly looks good to me, can you please add the screenshots in a table so that it is easy to compare?

**Comment by anandwana001** [comment]
Assign me once all test case gets pass.

**Comment by rishidyno** [comment]
Ptal @anandwana001   Assigning this to you.

File: app/src/main/res/values-land/dimens.xml [hunk 2]
```diff
@@ -229,6 +229,7 @@
   <dimen name="faq_single_padding_top">28dp</dimen>
   <dimen name="faq_single_padding_end">72dp</dimen>
   <dimen name="faq_single_padding_bottom">56dp</dimen>
+  <dimen name="faq_single_activity_scroll_padding_Bottom">108dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-land/dimens.xml]: Rename to `faq_single_activity_scroll_padding_bottom`

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 3]
```diff
@@ -250,6 +250,7 @@
   <dimen name="faq_single_padding_top">48dp</dimen>
   <dimen name="faq_single_padding_end">32dp</dimen>
   <dimen name="faq_single_padding_bottom">132dp</dimen>
+  <dimen name="faq_single_activity_scroll_padding_Bottom">0dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Rename to `faq_single_activity_scroll_padding_bottom`

File: app/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -409,6 +409,7 @@
   <dimen name="faq_single_padding_top">22dp</dimen>
   <dimen name="faq_single_padding_end">28dp</dimen>
   <dimen name="faq_single_padding_bottom">60dp</dimen>
+  <dimen name="faq_single_activity_scroll_padding_Bottom">80dp</dimen>
```
Comment (rt4914) [inline app/src/main/res/values/dimens.xml]: Rename to `faq_single_activity_scroll_padding_bottom`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@rishidyno You can notice in screenshots that the tablet-landscape screen has changed. Also looking at the code I would suggest you to merge only `layout` and `layout-land` file.
So in the end instead of three you will have only two files: `layout/faq_single_activity` and `layout-sw600dp-land/faq_single_activity`

**Comment by rishidyno** [comment]
@rt4914 @anandwana001 @prayutsu  I have done the required changes. Please review it.

File: .idea/deploymentTargetDropDown.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+          <Key>
+            <type value="VIRTUAL_DEVICE_PATH" />
+            <value value="C:\Users\RISHI RAJ\.android\avd\Pixel_C_API_30_1.avd" />
+          </Key>
+        </deviceKey>
+      </Target>
+    </targetSelectedWithDropDown>
+    <timeTargetWasSelectedWithDropDown value="2021-08-04T06:54:43.519797800Z" />
+  </component>
+</project>
```
Comment (prayutsu) [inline .idea/deploymentTargetDropDown.xml]: Revert these changes

**Review by prayutsu — CHANGES_REQUESTED** [review]
@rishidyno Left one comment, also when you add the suggested changes, then you should reply back in the conversation threads so that the reviewer is aware that you have made that particular change or you faced some problem.

File: app/src/main/res/values-sw600dp-land/booleans.xml (hunk 1 above)
Comment (prayutsu) [inline app/src/main/res/values-sw600dp-land/booleans.xml]: Remove this if we are not actually using this boolean anywhere.

File: app/src/main/res/layout/faq_single_activity.xml [hunk 6]
```diff
@@ -40,10 +41,12 @@
       <ScrollView
         android:layout_width="match_parent"
         android:layout_height="match_parent"
+        android:layout_gravity="center_horizontal"
         android:background="@color/oppiaGreyBackground"
         android:clipToPadding="false"
+        android:fillViewport="@bool/faq_single_activity_scroll_fill_view_port"
         android:overScrollMode="never"
```
Comment (prayutsu) [inline app/src/main/res/layout/faq_single_activity.xml:45]: We can probably remove this attribute now ( As we are keeping the tablet landscape layout)

**Review by prayutsu — CHANGES_REQUESTED** [review]
Left some more comments, PTAL @rishidyno

**Comment by rt4914** [comment]
@rishidyno Please reply to the open comments so that reviewer can know what exactly you did for that particular comment.

**Comment by rishidyno** [comment]
@rt4914 All addressed changes are done.  I have retained all the attributes needed and deleted unnecessary ones.

File: app/src/main/res/values-land/dimens.xml (hunk 2 above)
Comment (rishidyno) [reply app/src/main/res/values-land/dimens.xml]: Yes Sir I have renamed it to `faq_single_activity_scroll_padding_bottom` in dimens.xml-land

File: app/src/main/res/values-sw600dp-land/dimens.xml (hunk 3 above)
Comment (rishidyno) [reply app/src/main/res/values-sw600dp-land/dimens.xml]: This has also been renamed to `faq_single_activity_scroll_padding_bottom`  in dimens.xml sw600dp-land.

File: app/src/main/res/values/dimens.xml (hunk 4 above)
Comment (rishidyno) [reply app/src/main/res/values/dimens.xml]: same in dimens.xml

File: .idea/deploymentTargetDropDown.xml (hunk 5 above)
Comment (rishidyno) [reply .idea/deploymentTargetDropDown.xml]: Done

File: app/src/main/res/values-sw600dp-land/booleans.xml (hunk 1 above)
Comment (rishidyno) [reply app/src/main/res/values-sw600dp-land/booleans.xml]: Ya! I have removed this now unnecessary file.

File: app/src/main/res/layout/faq_single_activity.xml (hunk 6 above)
Comment (rishidyno) [reply app/src/main/res/layout/faq_single_activity.xml:45]: Done

**Comment by rishidyno** [comment]
> @rishidyno Please reply to the open comments so that reviewer can know what exactly you did for that particular comment.

Sir! sorry for the inconvenience. I will take care of this in the future.

**Comment by rishidyno** [comment]
@rt4914  Please review this PR

**Review by prayutsu — APPROVED** [review]
LGTM, thanks @rishidyno.

**Comment by rishidyno** [comment]
Ptal @rt4914   

 All tests have passed and I have updated the project.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Comment by rishidyno** [comment]
> LGTM, thanks.

Thanks @rt4914
